src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -2765,56 +2782,83 @@ private static int IndexOfMinMaxCore<TIndexOfMinMax>(ReadOnlySpan<float> x)
                     current = Vector512.LoadUnsafe(ref xRef, (uint)i);
                     curIndex += increment;
 
-                    nanMask = ~Vector512.Equals(current, current);
-                    if (nanMask != Vector512<float>.Zero)
+                    if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                     {
-                        return i + IndexOfFirstMatch(nanMask);
+                        nanMask = ~Vector512.Equals(current, current);
+                        if (nanMask != Vector512<T>.Zero)
+                        {
+                            return i + IndexOfFirstMatch(nanMask);
+                        }
                     }
 
                     TIndexOfMinMax.Invoke(ref result, current, ref resultIndex, curIndex);
 
-                    i += Vector512<float>.Count;
+                    i += Vector512<T>.Count;
                 }
 
                 // If any elements remain, handle them in one final vector.
                 if (i != x.Length)
                 {
-                    current = Vector512.LoadUnsafe(ref xRef, (uint)(x.Length - Vector512<float>.Count));
-                    curIndex += Vector512.Create(x.Length - i);
+                    current = Vector512.LoadUnsafe(ref xRef, (uint)(x.Length - Vector512<T>.Count));
+                    curIndex += CreateVector512T(x.Length - i);
 
-                    nanMask = ~Vector512.Equals(current, current);
-                    if (nanMask != Vector512<float>.Zero)
+                    if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                     {
-                        return curIndex[IndexOfFirstMatch(nanMask)];
+                        nanMask = ~Vector512.Equals(current, current);
+                        if (nanMask != Vector512<T>.Zero)
+                        {
+                            int indexInVectorOfFirstMatch = IndexOfFirstMatch(nanMask);
+                            return
+                                sizeof(T) == sizeof(long) ? (int)(long)(object)curIndex.As<T, long>()[indexInVectorOfFirstMatch] :
+                                sizeof(T) == sizeof(int) ? (int)(object)curIndex.As<T, int>()[indexInVectorOfFirstMatch] :
+                                throw new NotSupportedException();
```

Comment:
return `int.CreateTruncating(curIndex[IndexOfFirstMask(nanMask])`?

Reply:
As with the other case, I'm effectively smuggling integers in the Ts, using the corresponding integer type based on size. I can't just create truncating because the values will be interpreted incorrectly.